### PR TITLE
Upgrade to Scala 2.10.5 and 2.11.6.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Comment by sjrd** [comment]
Nightly: https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/1878/
Weekly: https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/1879/

**Comment by sjrd** [comment]
Review by @gzm0 one of these days.

Btw, I published the compiler for 2.10.5 and 2.11.6 in versions 0.5.6, 0.6.0 and 0.6.1.

**Comment by sjrd** [comment]
The 2.10.5 and 2.11.6 of nightly and weekly passed.

File: ci/checksizes.sh [hunk 1]
```diff
@@ -8,10 +8,10 @@ case $FULLVER in
   2.10.2)
     VER=2.10
     ;;
-  2.11.5)
+  2.11.6)
     VER=2.11
     ;;
-  2.10.3|2.10.4|2.11.0|2.11.1|2.11.2|2.11.4)
+  2.10.3|2.10.4|2.11.0|2.11.1|2.11.2|2.11.4|2.11.5)
```
Comment (gzm0) [inline ci/checksizes.sh]: Aren't we missing `2.10.5` here?

**Comment by gzm0** [comment]
That's all

File: ci/checksizes.sh (hunk 1 above)
Comment (sjrd) [inline ci/checksizes.sh]: Yes indeed.

**Comment by sjrd** [comment]
Updated.

**Comment by sjrd** [comment]
@gzm0 I have an issue with the CI infrastructure. See this failing worker task:
https://scala-webapps.epfl.ch/jenkins/job/scalajs-task-worker/31081/console

It reports a binary compatibility issue wrt 0.6.1, saying that `OptimizerCore.withState` changed return type from `Object` to `TailRec`. But that's not true, it hasn't. A local run reports no issue.

But `withState` is precisely the method I had to change between the failed attempt at releasing 0.6.1, and the actual 0.6.1. I guess this worker was the one testing `sbt-plugin-test`, and so it published locally the wrong `0.6.1`. Now, when trying to check bin compat, it resolves to the locally published, wrong, 0.6.1, and reports a binary compat issue which, in fact, does not exist.

Do you know how can I make this worker (and possibly others) forget about this wrong local 0.6.1?

**Comment by sjrd** [comment]
retest this please

**Comment by gzm0** [comment]
> Do you know how can I make this worker (and possibly others) forget about this wrong local 0.6.1?

You'll have to nuke the local directory in the ivy/sbt home at `/localhome/jenkins/scala-js-sbt-homes/sbt-home`. It would be best to have a home per Jenkins workspace and remove the local directory before each run. (So we could get caching, but not leak stuff between builds).

**Comment by sjrd** [comment]
I don't think I have access to these machines ^^

**Comment by sjrd** [comment]
OK, done. At least on that machine.

**Comment by gzm0** [comment]
> I don't think I have access to these machines ^^

Uname & Password are in the LAMP wiki (in case you haven't figured out). Or ask Fabien.

**Pull request merged**